website/src/pages/[organism]/api/sequences.ts

```diff
@@ -0,0 +1,185 @@
+import type { Readable } from 'stream';
+
+import type { APIRoute } from 'astro';
+import type { AxiosResponse } from 'axios';
+import { z } from 'zod';
+
+import { cleanOrganism } from '../../../components/Navigation/cleanOrganism.ts';
+import { LapisClient } from '../../../services/lapisClient.ts';
+import { ACCESSION_VERSION_FIELD } from '../../../settings.ts';
+import type { ProblemDetail } from '../../../types/backend.ts';
+
+type SearchParams = {
+    segment?: string;
+    headerFields: string[];
+    downloadFileBasename: string;
+    queryFilters: { [p: string]: string };
+};
+
+export const GET: APIRoute = async ({ params, request }) => {
+    const searchParams = getSearchParams(new URL(request.url));
+
+    const organism = cleanOrganism(params.organism);
+    if (organism.organism === undefined) {
+        return new Response(`Organism ${params.organism} not found`, { status: 404 });
+    }
+    const lapisClient = LapisClient.createForOrganism(organism.organism.key);
+
+    const fastaHeaderMapResult = await getAccessionVersionToFastaHeaderMap(lapisClient, searchParams);
+
+    if (fastaHeaderMapResult.isErr()) {
+        return Response.json(fastaHeaderMapResult.error, { status: fastaHeaderMapResult.error.status });
+    }
+
+    const { fastaHeaderMap, dataVersion: detailsDataVersion } = fastaHeaderMapResult.value;
+
+    const sequencesResponse = await lapisClient.streamSequences(searchParams.segment, {
+        ...searchParams.queryFilters,
+        dataFormat: 'ndjson',
+    });
+
+    const sequencesDataVersion = sequencesResponse.headers['lapis-data-version'];
+    if (sequencesDataVersion !== detailsDataVersion) {
```

Comment:
It would be better to retry (once?) in this case, but as a temporary solution, we can leave it as is.